_posts/2018-04-09-synching-sia.md

```diff
@@ -0,0 +1,131 @@
+
+## Future optimizations
+
+### Increased batch size
+
+Sia downloads blocks in batches of 10. That means Sia requests blocks at least 14,800 times for a full blockchain sync. I believe Sia could request blocks in batches of 25, or even 100 and reduce some of the overhead from making so many requests. I proposed that the devs raise the batch number to 25 but my proposal was denied. You can read the discussion and compelling arguments from both sides in my pull request [#2922](https://github.com/NebulousLabs/Sia/pull/2922).
+
+### A consensus overhaul
+
+Many optimizations such as parallel block downloads, smarter timeouts, I/O buffering, and header-first block retrieval could be added with a consensus package overhaul.  I don’t believe this is the best use of time for the limited dev resources Nebulous currently has, but I do believe an overhaul will be necessary if Sia and its blockchain continue to grow.
+
+Concerns about the blockchain size and performance have been brought up several times, and the Sia devs have several ideas on how to support a large blockchain.  You can read some of the discussion on GitHub in issue [#2897](https://github.com/NebulousLabs/Sia/issues/2897).
+
+## A 73-minute sync is mostly academic
+
+The reality is that most users are not going to be able to synchronize the blockchain in 73 minutes.  I challenged myself to see how fast I was able to do a full sync, and 73 minutes was the number I got.  Most users aren’t going to want to run Sia with so few modules loaded, or modify the source code and recompile.  Most users aren’t going to have an extra 11GB of RAM lying around to create a RAMDisk, and most users aren’t going to have 200+Mbps Internet connections.
+
+However, some users may be interested in a subset of the optimizations mentioned, perhaps they already run a custom version of Sia and can improve their performance with lower timeouts, or some users have lots of RAM, but are bottlenecked by the I/O performance of their hard drive.
+
+An excerpt from my `consensus.log` file showing Sia startup, and IBD (Initial Blockchain Download) complete 73 minutes later with the elapsed time in seconds spent on each step.  Link to [full log](https://gist.github.com/tbenz9/6130ca40b94c6550b62b2ba65a7d77c8).
+
+```text
+2018/04/05 20:11:41.546036 persist.go:73: STARTUP: Logging has started. Siad Version 1.3.2
+2018/04/05 20:11:49.321247 accept.go:275: Height 1000 at 1522959109
+… TRUNCATED
+2018/04/05 21:23:39.602232 accept.go:275: Height 148000 at 1522963419
+2018/04/05 21:23:39.602263 accept.go:276: Download time: 168.738052889
+2018/04/05 21:23:39.602270 accept.go:277: Validate time: 2631.783831114
+2018/04/05 21:23:39.602274 accept.go:278: Apply time: 250.713886247
+2018/04/05 21:23:39.602278 accept.go:279: Subscribers time: 0.01789937
+… TRUNCATED
+2018/04/05 21:24:57.406911 synchronize.go:642: INFO: IBD done, synced with 6 peers
+```
```

Comment:
Suggest cutting lines 107-120. It adds a lot of space, but I don't think it's that interesting for the reader. At most, I think we should have a "Raw data" section with simple links (like in the [load test posts](https://blog.spaceduck.io/load-test-2/#raw-data)).